app/Console/Commands/AchievementsCron.php

```diff
@@ -532,4 +535,26 @@ private function nThActivity($activityAmount) {
         }
         return $selected;
     }
+
+    private function percentageActivities($percentage) {
+        $selected = [];
+            if (Carbon::now()->day == 1) {
+                $users = User::all();
+                foreach ($users as $user) {
+                    $participated = ActivityParticipation::where('user_id', $user->id)->pluck('activity_id');
+                    $activities = Activity::whereIn('id', $participated)->pluck('event_id');
+                    $CountEventsParticipated = Event::whereIn('id', $activities)->where('end','>',Carbon::now()->subMonth()->valueOf())->where('end','<',Carbon::now()->valueOf())->where('secret','=', '0')->count();
+
+                    $possibleActivities = Event::where('end','>',Carbon::now()->subMonth()->valueOf())->where('end','<',Carbon::now()->valueOf())->where('secret','=', '0')->has('activity')->count();
+
+                    if (floor($CountEventsParticipated / $possibleActivities * 100) >= $percentage) {
```

Comment:
Maybe we want a minimum number of possible activities? otherwise you could get a divide by 0 error.